R/bs_sass.R

```diff
@@ -61,23 +61,34 @@ bootstrap <- function(theme = bs_theme_get(),
   )
   opts <- utils::modifyList(opts, options)
 
-  # Temp dir for building the HTML dependencies
-  output_path <- tempfile("bscustom")
-  dir.create(output_path)
+  cache_key <- sass::sass_cache_key(theme, opts)
 
-  # Compile sass in temp dir
-  output_css <- if (minified) "bootstrap-custom.min.css" else "bootstrap-custom.css"
-  sass::sass(
-    input = theme,
-    output = file.path(output_path, output_css),
-    write_attachments = TRUE,
-    options = opts,
-    ...
+  # Temp dir for building the HTML dependencies
+  output_path <- file.path(
+    tempdir(),
+    "bootstraplib",
+    paste0("bootstrap-", cache_key)
```

Comment:
@schloerke I vaguely remember you running into learnr issues with filenames that were too long -- could that be an issue here?

Reply:
IDK if it matters, but the hash isn't too long. For example, the dir will look like `bootstrap-52a853bb558aa1b2`.

Reply:
That should be fine. For learnr it was like a few hundred characters on windows for some reason.  But that was because when normalized on a university machine, it resolved to a path that was really convoluted.  I also think for the learnr situation it was caused by paste'ing two paths together.  For now, I wouldn't worry about it. 

If anything you could drop the `bootstrap-` prefix on the file as it is already namespaced by the `bootstrap` folder.